src/analyz/to_json.rs

```diff
@@ -133,10 +155,13 @@ impl<'tcx> TraitInst<'tcx> {
     pub fn dyn_ty(&self, tcx: TyCtxt<'tcx>) -> Option<ty::Ty<'tcx>> {
         let trait_ref = self.trait_ref?;
         let mut preds = Vec::with_capacity(self.projs.len() + 1);
-        preds.push(ty::ExistentialPredicate::Trait(trait_ref));
-        preds.extend(self.projs.iter().map(|p| ty::ExistentialPredicate::Projection(*p)));
-        let preds = tcx.intern_existential_predicates(&preds);
-        Some(tcx.mk_dynamic(ty::Binder::bind(preds), tcx.mk_region(ty::RegionKind::ReErased)))
+        preds.push(ty::Binder::dummy(ty::ExistentialPredicate::Trait(trait_ref)));
+        preds.extend(
+            self.projs.iter().map(|p| ty::Binder::dummy(ty::ExistentialPredicate::Projection(*p))),
+        );
+        let preds = tcx.intern_poly_existential_predicates(&preds);
+        // TO REVIEW: DynKind?
+        Some(tcx.mk_dynamic(preds, tcx.mk_region(ty::RegionKind::ReErased), DynKind::Dyn))
```

Comment:
`DynKind` exists to distinguish normal `dyn Trait` from the experimental `dyn* Trait` type.  I don't think anybody really uses `dyn*` at the moment, so it's okay if we don't support it.

```suggestion
        // Always emit `DynKind::Dyn`.  We don't support `dyn*` (`DynKind::DynStar`) yet.
        Some(tcx.mk_dynamic(preds, tcx.mk_region(ty::RegionKind::ReErased), DynKind::Dyn))
```